File: 151-ReverseWordsInAString/151-ReverseWordsInAString.java
// Last updated: 16/10/2025, 04:28:49
class Solution {
    public String reverseWords(String s) {

        //we can use "\s+" also instead of " +" to match any number of whitespaces.
        String[] word = s.split(" +"); 
        StringBuilder res = new StringBuilder("");

        for(int i=word.length-1;i>=0;i--){
            res = res.append(word[i] + " ");
        }
        return res.toString().trim();
    }
}

File: 151-ReverseWordsInAString/151-ReverseWordsInAString.cpp
// Last updated: 16/10/2025, 04:28:55
class Solution {
public:
    string reverseWords(string s) {
        int n = s.length();
        stack<string> st;
        string str1 = "";
        for(int i=0;i<n;i++){
            if(s[i] == ' '){
                if(str1.length() > 0)st.push(str1);
                str1 = "";
                continue;
            }
            str1 += s[i];
            
        }
        if(str1.length() > 0){
            st.push(str1);
        }
        
        
        string res = "";
        while(!st.empty()){
            res += st.top();
            st.pop();
            if(!st.empty()){
                res += ' ';
            } 
        }
        return res;
    }
};